src/engine.ts

```diff
@@ -57,14 +66,39 @@ export class Workflows {
     return new Workflows(docs, repository);
   }
 
-  static validate(obj: any): obj is Workflow {
-    return typeof obj === 'object' && obj.version && obj.name && Array.isArray(obj.steps);
+  static parse(w: any): Workflow {
+    if (w) {
+      w.on = parseTrigger(w.on);
+    }
+    return w;
+  }
+
+  static validate(obj: Workflow): boolean {
```

Comment:
こちらは type guard 関数に戻したいです．
```suggestion
  static validate(obj: any): obj is Workflow
```